### PR TITLE
Basic test for the PileUp

### PR DESCRIPTION
**Pull request by pleroy**

### CODE REVIEW

File: ksp_plugin/pile_up.hpp [hunk 1]
```diff
@@ -33,7 +33,7 @@ using quantities::Mass;
 // A |PileUp| handles a connected component of the graph of |Parts| under
 // physical contact.  It advances the history and prolongation of its component
 // |Parts|, modeling them as a massless body at their centre of mass.
-class PileUp final {
+class PileUp {
```
Comment (eggrobin) [inline ksp_plugin/pile_up.hpp]: Why not final? In that case, have a virtual destructor.

File: ksp_plugin/pile_up.hpp [hunk 2]
```diff
@@ -72,9 +71,9 @@ class PileUp final {
       Ephemeris<Barycentric>::AdaptiveStepParameters const&
           adaptive_step_parameters);
 
-  // Adjusts the |Bubble| degrees of freedom of all parts in this pile up based
-  // on the origin of |Bubble| given in |Barycentric|, and on the |RigidPileUp|
-  // degrees of freedom of the parts (as set by |DeformPileUpIfNeeded|).
+  // Adjusts the degrees of freedom of all parts in this pile up based on the
+  // position of the pile-up computed by |AdvanceTime| and on the |RigidPileUp|
```
Comment (eggrobin) [inline ksp_plugin/pile_up.hpp]: position? isn't this degrees of freedom?

File: ksp_plugin_test/pile_up_test.cpp [hunk 3]
```diff
@@ -13,105 +16,469 @@ namespace ksp_plugin {
...
+                                       Displacement<ApparentBubble>(
+                                           {2.0 * Metre,
+                                            0.0 * Metre,
+                                            -2.0 / 3.0 * Metre}), 0),
+                      AlmostEquals(Velocity<ApparentBubble>(
+                                       {20.0 * Metre / Second,
+                                        0.0 * Metre / Second,
+                                        -20.0 / 3.0 * Metre / Second}), 0)));
+
+    pile_up.DeformPileUpIfNeeded();
```
Comment (eggrobin) [inline ksp_plugin_test/pile_up_test.cpp]: This should not be part of a function that checks invariants.

File: ksp_plugin/pile_up.hpp (hunk 1 above)
Comment (pleroy) [inline ksp_plugin/pile_up.hpp]: Because of `TestablePileUp`.  Done.

File: ksp_plugin/pile_up.hpp (hunk 2 above)
Comment (pleroy) [inline ksp_plugin/pile_up.hpp]: Done.

File: ksp_plugin_test/pile_up_test.cpp (hunk 3 above)
Comment (pleroy) [inline ksp_plugin_test/pile_up_test.cpp]: Done.

**Pull request merged**